validphys2/src/validphys/dataplots.py

```diff
@@ -891,54 +891,82 @@ def plot_smpdf(pdf, dataset, obs_pdf_correlations, mark_threshold: float = 0.9):
 
     plotting_var = info.get_xcol(table)
 
-    # TODO: vmin vmax should be global or by figure?
-    vmin, vmax = min(plotting_var), max(plotting_var)
-    if info.x_scale == 'log':
-        norm = mcolors.LogNorm(vmin, vmax)
+    categorical = not np.issubdtype(plotting_var.dtype, np.number)
+    if categorical:
+        # Categorical
```

Comment:
Maybe explain that this means discrete colouring, the comment alone is not very enlightening.